src/controller/authController.ts

```diff
@@ -22,6 +26,44 @@ export namespace Auth
         }
     }
     
+    export async function getRegister(req: Request, res: Response): Promise<void> {
+        res.locals.roles = await authService.getRoles();
+
+        res.render("register")
+    }
+    
+    export async function postRegister(req: Request, res: Response): Promise<void> {
+        let data: Register = req.body;
+        
+        const rerender = async (error: string) => {
+            res.locals.errorMessage = error;
+            res.locals.roles = await authService.getRoles();
+            res.locals = {...res.locals, ...req.body};
+
+            res.render("register");
+        }
+        
+        const emailError = registerValidator.validateEmail(data.email);
+        if (emailError) {
+            await rerender(`Please check your details: ${emailError}`);
+            return;
+        }
+        
+        const passwordError = registerValidator.validatePassword(data.password);
+        if (passwordError) {
+            await rerender(`Please check your details: ${passwordError}`);
+            return;
+        }
+
+        try {
+            await authService.register(data);
```

Comment:
You should probably validate the role as well. At the moment I can pretty much submit whatever I want in that field